### DISCUSSION TITLE
setTimeout for animation no longer works

### DISCUSSION THREAD

**Discussion by WadeMorrison** [google/model-viewer#3341] [Q&A] [answered]
Hey there! I saw there was an update a couple days ago, and my "Open/Close" animation no longer works properly. Not sure if I need to change some of the script or not? I basically had a 1 second animation, and had it "pause" at 500ms when clicked on an "Open/Close" button. And when clicked again, it would resume the animation for another 500ms (creating a loop-able open/close animation on some cabinets). Now it will pause after 500ms in the "Open" position, but when clicked again, it jumps back and opens again. Instead of playing the rest of the animation and "closing" it.

The code I have goes like:

```
const OpenCloseBtn = () => {
    if (!product.hasAnim) return <></>;
        const doorOpen = () => {
        const animLength = 500; //ms
        const modelViewer = document.querySelector("model-viewer");
            modelViewer.animationName = 'OpenClose';
            modelViewer.play();
            setTimeout(()=>{modelViewer.pause();}, animLength);  
            console.log("Playing Animation")
};

return (
    <div className="openCloseBtn">
        <button onClick={doorOpen} className="openBtn">Open/Close</button>
    </div>
    );
};
```
Any help would be greatly appreciated!

**Comment by clochardM33** [comment, marked as answer]
Have a look at
https://github.com/google/model-viewer/issues/3326
It seems to be in the current release version and not your bit.
